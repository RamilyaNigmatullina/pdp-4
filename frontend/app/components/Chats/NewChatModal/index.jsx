import React from 'react';
import Modal from 'react-bootstrap/Modal';
import PropTypes from 'prop-types';
import { fetchUsersWithoutChat } from '../api/index';
import styles from './styles.module.scss';

class NewChatModal extends React.Component {
  state = {
    users: null,
  }

  componentDidUpdate() {
    if (this.props.isShown && !this.state.users) this.fetchUsers();
    if (!this.props.isShown && this.state.users) this.setState({ users: null });
  }

  renderUserRow = (user) => (
    <div className={styles.userItem} key={user.id} onClick={(e) => this.props.onChatCreate(e, user.id)}>
      { user.full_name }
    </div>
  );

  render() {
    const { onClose, isShown } = this.props;
    const { users } = this.state;

    if (!isShown) return null;

    return (
      <Modal aria-labelledby="contained-modal-title-vcenter" onHide={onClose} show={isShown} size="md" centered >
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            Direct Messages
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          { users && !!users.length && this.state.users.map((user) => this.renderUserRow(user)) }
          { users && !users.length && <div>No available users</div> }
        </Modal.Body>
      </Modal>
    );
  }

  fetchUsers = () => {
    fetchUsersWithoutChat().then((data) => this.setState({ users: data }));
  }
}

NewChatModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  isShown: PropTypes.bool.isRequired,
  onChatCreate: PropTypes.func.isRequired,
};

export default NewChatModal;
