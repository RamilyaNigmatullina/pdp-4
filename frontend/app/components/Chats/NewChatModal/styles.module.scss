.userItem {
  align-items: center;
  cursor: pointer;
  display: flex;
  height: 50px;
  padding: 0 10px;
  width: 100%;

  &:hover {
    background-color: #ebebeb;
  }
}
