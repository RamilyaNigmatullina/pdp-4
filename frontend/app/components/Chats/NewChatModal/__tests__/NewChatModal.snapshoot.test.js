import React from 'react';
import { shallow } from 'enzyme';
import NewChatModal from '../index';

describe('NewChatModal', () => {
  let props;

  const render = () => shallow(<NewChatModal {...props} />);

  beforeEach(() => {
    props = {
      onClose: jest.fn(),
      isShown: true,
      onChatCreate: jest.fn(),
    };
  });

  it('renders new chat modal', () => {
    const component = render();

    expect(component).toMatchSnapshot();
  });

  describe('when is not shown', () => {
    beforeEach(() => {
      props.isShown = false;
    });

    it('renders new chat modal', () => {
      const component = render();

      expect(component).toMatchSnapshot();
    });
  });
});
