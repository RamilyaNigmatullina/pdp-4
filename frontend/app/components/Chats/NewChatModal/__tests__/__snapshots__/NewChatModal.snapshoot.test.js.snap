// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewChatModal renders new chat modal 1`] = `
<Bootstrap(Modal)
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  onHide={[MockFunction]}
  show={true}
  size="md"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      id="contained-modal-title-vcenter"
    >
      Direct Messages
    </ModalTitle>
  </ModalHeader>
  <ModalBody />
</Bootstrap(Modal)>
`;

exports[`NewChatModal when is not shown renders new chat modal 1`] = `""`;
